src/helpers/NodeHelper.ts

```diff
@@ -110,4 +117,59 @@ export class NodeHelper {
     }
     return allURIs;
   };
+
+  /**
+   * iterate through all the nodes we have with a chainId check.
+   * - log the failing nodes
+   * - 3 fails in < 15 minutes sends the node to the invalidNodes list
+   * returns an Array of working mainnet nodes
+   */
+  static checkAllNodesStatus = async () => {
+    let workingURI: string = "none";
+    var promises: any[] = [];
+    NodeHelper.getNodesUris().forEach(URI => {
```

Comment:
you could clean this up by doing `promises =  NodeHelper.getNodeUris().map(...)` and using the async/await notation to return the promise rather than mixing the `then`.